compiler/rustc_next_trait_solver/src/canonicalizer.rs

```diff
@@ -19,17 +19,28 @@ const NEEDS_CANONICAL: TypeFlags = TypeFlags::from_bits(
 )
 .unwrap();
 
+#[derive(Debug, Clone, Copy)]
+enum CanonicalizeInputKind {
+    /// When canonicalizing the `param_env`, we keep `'static` as merging
+    /// trait candidates relies on it when deciding whether a where-bound
+    /// is trivial.
+    ParamEnv,
```

Comment:
new solver special cases `'static`? I thought we weren't supposed to do this :thinking:

Reply:
We need to do so for the `param_env` to detect "global where-bounds". See https://github.com/rust-lang/rust/pull/133643 / the FCP in https://github.com/rust-lang/rust/pull/132325

Reply:
ah yh that makes sense